src/app/core/services/label.service.ts

```diff
@@ -109,17 +118,27 @@ export class LabelService {
     this.labelColorMap.clear();
   }
 
-  /**
-   * Converts the (color) hex value into RGB format
-   * @param hex: the hex value
+   /**
+   * Returns true if the given color is considered "dark"
+   * The color is considered "dark" if its luminance is less than COLOR_DARKNESS_THRESHOLD
+   * @param inputColor: the color
    */
-  hexToRgb(hex: string) {
-    const rgbResult = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
-    return rgbResult ? {
-      r: parseInt(rgbResult[1], 16),
-      g: parseInt(rgbResult[2], 16),
-      b: parseInt(rgbResult[3], 16)
-    } : null;
+  isDarkColor(inputColor: string): boolean {
+    const color = (inputColor.charAt(0) === '#') ? inputColor.substring(1, 7) : inputColor;
+    const r = parseInt(color.substring(0, 2), 16);
+    const g = parseInt(color.substring(2, 4), 16);
+    const b = parseInt(color.substring(4, 6), 16);
+    const rgb = [r / 255, g / 255, b / 255];
+    const c = rgb.map((col) => {
```

Comment:
while r, g and b is understandable, I feel we can use a better name than c?

Reply:
As they are constants, should they be `R`, `G`, ...?